ci_scripts/check_installability_plugins.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+"""
+Check if all Hub-listed plugins can be installed together with latest napari
+"""
+
+import json
+from subprocess import CalledProcessError, check_output, STDOUT
+
+import requests
+
+NPE2API_CONDA = "https://npe2api.vercel.app/api/conda"
+
+
+def check_output_verbose(*args, **kwargs):
+    """
+    Check return code of a subprocess, capturing stdout and stderr.
+    If an error happened, print both streams, then raise.
+    """
+    kwargs.pop("stderr", None)
+    kwargs.pop("text", None)
+    kwargs.pop("universal_newlines", None)
+    try:
+        return check_output(*args, stderr=STDOUT, text=True, **kwargs)
+    except CalledProcessError as exc:
+        print("Output:")
+        print(exc.stdout)
+        raise
+
+
+def latest_napari_on_conda_forge():
+    r = requests.get("https://api.anaconda.org/package/conda-forge/napari")
+    r.raise_for_status()
+    return r.json()["latest_version"]
+
+
+def all_plugin_names():
+    r = requests.get(NPE2API_CONDA)
+    r.raise_for_status()
+    return r.json()
+
+
+def is_latest_version(name, version):
+    r = requests.get(f"{NPE2API_CONDA}/{name}")
+    r.raise_for_status()
+    latest_version = r.json()["latest_version"]
+    return version >= latest_version
+
+
+def solve(specs):
+    resp = check_output_verbose(
+        [
+            "micromamba",
+            "create",
+            "-n",
+            "notused",
+            "--dry-run",
+            "-c",
+            "conda-forge",
+            "--json",
+            *specs,
+        ]
+    )
+    print(resp)
```

Comment:
Can we check that pyqt is not part of the solution :) ?

Reply:
Not really. Some dependencies (direct or secondary) might still pull it because their metadata said so. In a future `constructor` release I want to restore the `exclude` config key though, so we will be able to clear that out (but the environment will be inconsistent 😬). What we'll do is to export the correct `QT_API` env var so `qtpy` uses the bindings we want.

Reply:
Re the "excluded package will make environment inconsistent", I think we can always provide empty `pyqt`s in a custom channel so it shadows conda-forge's. Maybe we can also do something locally, mocking a cached channel, or an installed shim, or something.

Reply:
But, the idea would be able to detect those cases and ensure those plugins are not directly using pyqt? or are we talking about other dependencies and not plugins?

Reply:
It's more about the 2nd-order (and higher) dependencies. E.g. deps of plugins, or deps of deps of deps.